newrelic/python_agent.go

```diff
@@ -0,0 +1,71 @@
+package newrelic
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/libcnb"
+	"github.com/paketo-buildpacks/libpak"
+	"github.com/paketo-buildpacks/libpak/bard"
+	"github.com/paketo-buildpacks/libpak/effect"
+	"github.com/paketo-buildpacks/libpak/sherpa"
+)
+
+type PythonAgent struct {
+	BuildpackPath    string
+	ApplicationPath  string
+	Executor         effect.Executor
+	LayerContributor libpak.DependencyLayerContributor
+	Logger           bard.Logger
+}
+
+func NewPythonAgent(applicationPath string, buildpackPath string, dependency libpak.BuildpackDependency, cache libpak.DependencyCache) (PythonAgent, libcnb.BOMEntry) {
+	contributor, entry := libpak.NewDependencyLayer(dependency, cache, libcnb.LayerTypes{
+		Launch: true,
+	})
+	return PythonAgent{
+		ApplicationPath:  applicationPath,
+		BuildpackPath:    buildpackPath,
+		Executor:         effect.NewExecutor(),
+		LayerContributor: contributor,
+	}, entry
+}
+
+func (p PythonAgent) Contribute(layer libcnb.Layer) (libcnb.Layer, error) {
+	p.LayerContributor.Logger = p.Logger
+
+	p.Logger.Bodyf("Installing to %s", layer.Path)
+
+	fmt.Println("Installing New Relic Python Agent using pip3")
+	if err := p.Executor.Execute(effect.Execution{
+		Command: "pip3",
+		Args:    []string{"install", "newrelic==" + os.Getenv("NEWRELIC_PYTHON_VERSION")},
```

Comment:
You want to use ConfigurationResolver here to get the version. https://github.com/paketo-buildpacks/libpak/blob/5cea674db0f4bbb4c40352cd4ac18a11d5020f67/buildpack.go#L381

This will take the settings from buildpack.toml & allow you to consistently apply a default.

What we typically do is create the ConfigurationResolver in `build.go` (there is probably one already), resolve the variable you want there and then pass the value into `NewPythonAgent`. That way your agent doesn't need to worry about how the variable is resolved. It makes testing easier.